### DISCUSSION TITLE
How to override the internal calendar event type?

### DISCUSSION THREAD

**Discussion by stuffyAI** [bruceharrison1984/Schedulely#79] [General]
I am using my own event class and am curious on how to supply additional data to the event objects, my events array I pass in <Schedulely events={_said array_}> will always lose additional data as they are not defined in the InternalCalendarEvent.ts .

Any help is appreciated.

**Comment by stuffyAI** [comment]
Right now I pass my data into events as `summary = JSON.stringify(DATA)` and retrieve it in my custom event with JSON.parse(event.summary), but that is probably not the proper way to do it

**Reply by bruceharrison1984** [reply]
The current implementation is fairly restricted due to TS types. It should probably be refactored at this point so the Internal event is exposed, and can accept a generic argument so you can easily pass additional parameters into the component.

You current method should work until then. The internal transformation of the event makes it impossible to simply pass in a custom object because all the additional fields will be thrown away.

TLDR;
This area needs to be rewritten and cleaned up so this works better. I'll earmark this for the next version.

**Reply by bruceharrison1984** [reply]
I've created an issue to track this: https://github.com/bruceharrison1984/Schedulely/issues/80

**Reply by stuffyAI** [reply]
Thanks for the quick response, your library is very beginner friendly, thanks for making it!

**Reply by bruceharrison1984** [reply]
I'm glad you like it. I've started work on this already, so hopefully I can cut a new release by the end of the week or next.

**Reply by bruceharrison1984** [reply]
New version is up:

Changelog: https://github.com/bruceharrison1984/Schedulely/releases/tag/v0.2.3
Docs: https://bruceharrison1984.github.io/Schedulely/docs/Usage/CustomEventData

I had some down-time today and this was fairly easy to implement. It'll likely need to be cleaned up later, but it gets you what you need and is non-breaking.

**Reply by stuffyAI** [reply]
> I had some down-time today and this was fairly easy to implement. It'll likely need to be cleaned up later, but it gets you what you need and is non-breaking.

Thank you very much, this is exactly what I needed. Keep up the great work!
